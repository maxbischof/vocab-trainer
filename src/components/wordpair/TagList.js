import React from 'react'
import styled from 'styled-components'

export default function TagList({ tags }) {
  const colors = {
    Substantiv: 'var(--noun)',
    Verb: 'var(--verb)',
    Adjektiv: 'var(--adjective)',
  }
  const color = colors[tags.wordClass]

  return (
    <StyledTagList wordClass={tags.wordClass}>
      {tags.wordClass && <Tag color={color}>{tags.wordClass}</Tag>}
      {tags.gender && <Tag color={color}> {tags.gender}</Tag>}
      {tags.number && <Tag color={color}> {tags.number}</Tag>}
      {tags.person && <Tag color={color}> {tags.person}</Tag>}
    </StyledTagList>
  )
}

const StyledTagList = styled.div`
  margin: 10px 0;
  display: flex;
`

const Tag = styled.span`
  font-size: 80%;
  border: 1px solid ${(props) => `rgb(${props.color})`};
  color: ${(props) => `rgb(${props.color})`};
  border-radius: 3px;
  padding: 5px 5px;
  margin-right: 10px;
`
