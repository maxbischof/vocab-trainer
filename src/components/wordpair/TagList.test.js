import React from 'react'
import TagList from './TagList'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tags = {
    wordClass: 'Substantiv',
    gender: 'Maskulin',
    number: 'plural',
    person: '1. Person',
  }
  const tree = renderer.create(<TagList tags={tags} />).toJSON()
  expect(tree).toMatchSnapshot()
})
