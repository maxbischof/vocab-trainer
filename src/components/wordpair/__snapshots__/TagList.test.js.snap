// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ cBNdlU"
>
  <span
    className="sc-gsTCUz gXLnSw"
    color="var(--noun)"
  >
    Substantiv
  </span>
  <span
    className="sc-gsTCUz gXLnSw"
    color="var(--noun)"
  >
     
    Maskulin
  </span>
  <span
    className="sc-gsTCUz gXLnSw"
    color="var(--noun)"
  >
     
    plural
  </span>
  <span
    className="sc-gsTCUz gXLnSw"
    color="var(--noun)"
  >
     
    1. Person
  </span>
</div>
`;
